client/src/panels/panel_declarations/finances/vote_stat/in_year_vote_stat_breakdown.js

```diff
@@ -76,54 +76,45 @@ const planned_vote_or_stat_render = vs => function ({ calculations, footnotes, s
 
   const col = "{{est_in_year}}_estimates";
   const top_10_rows = _.take(data, 10);
-  const complement_amt = _.last(data)[col];
   const total_amt = d3.sum(data, _.property(col));
-  const column_names = {
-    name: text_maker("org"),
-    voted_stat: text_maker(isVoted ? "voted" : "stat"),
-    amount: text_maker("authorities"),
-  };
-  const amt_type = window.is_a11y_mode ? "compact1_written" : "compact1";
-  const total = { amount: amt_type} ;
-
-  const rows = _.map(top_10_rows, obj => {
-    const subj = Subject.Dept.lookup(obj.dept);
-    const display_values = {
-      name: <a href={infograph_href_template(subj)}> {subj.name} </a>,
-      voted_stat: obj.desc,
-      amount: <Format type={amt_type} content={obj[col]} />,
-    };
-    const search_values = {
-      name: subj.name,
-      voted_stat: obj.desc,
-    };
-    const sort_values = {
-      name: subj.name,
-      voted_stat: obj.desc,
-      amount: obj[col],
-    };
-    return {
-      display_values: display_values,
-      sort_values: sort_values,
-      search_values: search_values,
-    };
-  });
-  const complement_desc = text_maker(isVoted ? 'all_other_voted_items' : 'all_other_stat_items');
-  const rows_with_complement_amt = _.concat(rows, [{
-    display_values: {
-      name: "",
-      voted_stat: complement_desc,
-      amount: <Format type={amt_type} content={complement_amt} />,
+  
+  const subj_map = _.chain(top_10_rows)
+    .map( obj => [Subject.Dept.lookup(obj.dept).name, infograph_href_template(Subject.Dept.lookup(obj.dept))] )
+    .fromPairs()
+    .value();
```

Comment:
Mapping subjects by name here and multiple other places with these `map`. I think these are bad and potentially dangerous but not sure what to do. I can put dept code as the value and use that but I don't really like having to define sort, search every time..

Reply:
Hmm, I'm puzzled on `Dept` subject with `lookup_by_name` function though. What're you suggesting with this one?